### PR TITLE
Initial Morello build for capability-aware BDWGC

### PR DESCRIPTION
**Pull request by jeremysinger**
build Boehm collector for Morello

### CODE REVIEW

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
@jeremysinger If you click on "buildbot/capablevms-test-script" above you can see what buildbot did -- you probably want to click on "shell" on the next page. At the moment it's going to be a bit difficult to *test* the fork on CHERI because it crashes for anything big AIUI, but perhaps some simple tests could be added?

**Comment by jeremysinger** [comment]
bors try

File: doc/README.morello [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (ltratt) [inline doc/README.morello:1]: We can drop the extraneous blank lines at the top.

File: doc/README.morello [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+We use cmake for building bdwgc for Morello. We cross-compile
+on Linux using the standard Morello SDK that is configured and installed
+with cheribuild, i.e.
+`python3 cheribuild.py run-morello-purecap -d`
+
+In a directory ~/bdwgc_collector_experiment, containing
+empty subdirectories bdwgc-build and install, and the git repo
+in directory bdwgc, we run cmake as follows:
+
+
```
Comment (ltratt) [inline doc/README.morello:14]: Double blank line.

File: doc/README.morello [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+$ cmake -B bdwgc-build -S bdwgc -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=${HOME}/bdwgc_collector_experiment/install -DCMAKE_TOOLCHAIN_FILE=morello-purecap.cmake -Denable_gcj_support=OFF -Denable_parallel_mark=OFF -Denable_threads=OFF -Denable_dynamic_loading=OFF -G Ninja
+
+$ cmake --build bdwgc-build  && cmake --install bdwgc-build
```
Comment (ltratt) [inline doc/README.morello:17]: Double space before `&&` I think?

**Comment by ltratt** [comment]
AIUI building on Morello works pretty much out of the box which is good! I have a few minor formatting nits (I can't help myself!) but nothing substantive.

@djichthys can you also give this PR a quick once-over please? Thanks!

File: doc/README.morello (hunk 1 above)
Comment (jeremysinger) [reply doc/README.morello:2]: fixed in 1e441259371e1d238b0000424a4bc025b4408c2f

**Comment by ltratt** [comment]
Please squash.

**Comment by djichthys** [comment]
I think we should clean up the RISC-V code and then rebuild for Morello. So another function in `.buildbot.sh` ?

**Comment by jeremysinger** [comment]
> I think we should clean up the RISC-V code and then rebuild for Morello. So another function in `.buildbot.sh` ?

In commit 6f94fe58fd6f0b775708607e803651f01b407c10

**Comment by djichthys** [comment]
> > I think we should clean up the RISC-V code and then rebuild for Morello. So another function in `.buildbot.sh` ?
> 
> In commit [6f94fe5](https://github.com/capablevms/bdwgc/commit/6f94fe58fd6f0b775708607e803651f01b407c10)

Gave it a quick once over. Seems OK to me @jeremysinger @ltratt .

**Comment by ltratt** [comment]
Please squash.

**Comment by jeremysinger** [comment]
bors try

**Comment by jeremysinger** [comment]
bors try

**Comment by jeremysinger** [comment]
ready for merge @ltratt

**Pull request merged**